pallets/providers/src/utils.rs

```diff
@@ -1732,27 +1732,29 @@ impl<T: pallet::Config> MutateBucketsInterface for pallet::Pallet<T> {
         bucket_id: &Self::BucketId,
         delta: Self::StorageDataUnit,
     ) -> DispatchResult {
-        let (msp_id, user_id) = Buckets::<T>::try_mutate(&bucket_id, |maybe_bucket| {
+        Buckets::<T>::try_mutate(&bucket_id, |maybe_bucket| {
             let bucket = maybe_bucket.as_mut().ok_or(Error::<T>::BucketNotFound)?;
 
-            bucket.size = bucket.size.saturating_add(delta);
+            // Get the MSP ID and the user owner of the bucket
+            let msp_id = bucket
+                .msp_id
+                .ok_or(Error::<T>::BucketMustHaveMspForOperation)?;
+            let user_id = bucket.user_id.clone();
```

Comment:
Why when increasing we error out if there is no MSP, but when decreasing we don't error out, but just don't update the payment stream. It seems like the behaviour of `decrease_bucket_size` is the correct one, since there could be a bucket without an MSP, but if that is the case, we just don't update the payment stream.

Reply:
I might be wrong since I'm not the author of that logic, but I believe the idea behind this implementation was:
- A bucket can **_decrease_** in size if its owner submits an extrinsic to delete a specific file of that bucket and, in that extrinsic, includes a forest inclusion proof that proves that the file is part of that bucket. This flow can happen even if the bucket does not currently have an MSP and we shouldn't error out.
- On the other hand, the only way for a bucket to **_increase_** in size is for its MSP to accept a storage request, so increasing a bucket's size without it being currently stored by an MSP is an error, it should never happen.

Reply:
@ffarall just implemented the change that we talked about in the daily: buckets can now increase in size without having a MSP (or at least it's no longer enforce at the Providers level, the File System should be the one enforcing that if needed). Will merge when CI passes